test/test_invalid_headers.py

```diff
@@ -256,8 +284,12 @@ def test_matching_authority_host_headers(self,
             (b':method', b'GET'),
             (b'host', b'example.com'),
         ]
-        assert headers == h2.utilities.validate_headers(
-            headers, hdr_validation_flags)
+        if not hdr_validation_flags.is_trailer:
```

Comment:
This fixes the failure condition but raises a question of if we really want to use `hdr_validation_combos` with this test. This will never be a valid trailer header and we're now checking for trailers with pseudo headers elsewhere, so this seems a bit redundant.

If this is a special case that requires all the checks though, I like this over a blanket `try`/`except ProtocolError` catch since this should be the only condition we fail under.